app/src/main/kotlin/com/wire/android/ui/calling/incoming/IncomingCallViewModel.kt

```diff
@@ -31,49 +23,43 @@ import javax.inject.Inject
 class IncomingCallViewModel @Inject constructor(
     savedStateHandle: SavedStateHandle,
     private val navigationManager: NavigationManager,
-    private val conversationDetails: ObserveConversationDetailsUseCase,
-    private val allCalls: GetAllCallsUseCase,
+    private val incomingCalls: GetIncomingCallsUseCase,
     private val rejectCall: RejectCallUseCase,
     private val acceptCall: AnswerCallUseCase,
     private val callRinger: CallRinger
 ) : ViewModel() {
-    var callState by mutableStateOf(IncomingCallState())
-        private set
 
     val conversationId: ConversationId = savedStateHandle.get<String>(EXTRA_CONVERSATION_ID)!!.parseIntoQualifiedID()
 
     init {
         viewModelScope.launch {
             callRinger.ring(R.raw.ringing_from_them)
-            launch {
-                conversationDetails(conversationId = conversationId)
-                    .collect { initializeScreenState(conversationDetails = it) }
-            }
             launch {
                 observeIncomingCall()
             }
         }
     }
 
     private suspend fun observeIncomingCall() {
-        allCalls().collect {
-            if (it.first().conversationId == conversationId)
-                when (it.first().status) {
-                    CallStatus.CLOSED -> onCallClosed()
-                    else -> print("DO NOTHING")
-                }
+        incomingCalls().collect { calls ->
+            calls.find { call -> call.conversationId == conversationId }.also {
+                if (it == null)
+                    onCallClosed()
```

Comment:
I'm not sure about that one, it may cause the issue. 
when user accepts the call `navigationManager.navigateBack()` could be called twice: 
in `acceptCall()` and in `onCallClosed()`

Reply:
good catch.. I think this could not happen because the viewModelScope is being clearing before onCallClosed called..For safety, I will clear the job once we accept or decline a call